chainer/functions/loss/discriminative_loss.py

```diff
@@ -0,0 +1,378 @@
+import chainer
+from chainer.backends import cuda
+from chainer.functions.array.broadcast import broadcast_to
+from chainer.functions.array.cast import cast
+from chainer.functions.array.concat import concat
+from chainer.functions.array.expand_dims import expand_dims
+from chainer.functions.array.reshape import reshape
+from chainer.functions.array.stack import stack
+from chainer.functions.array.transpose import transpose
+from chainer.functions.math.average import average
+from chainer.functions.math.basic_math import absolute
+from chainer.functions.math.maximum import maximum
+from chainer.functions.math.sum import sum
```

Comment:
Python has `sum` as a built-in function. So I think it is better not to use the name and to hide the built-in function.